src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/color.test.ts

```diff
@@ -7,115 +7,224 @@
+
+          expect(() => {
+            colorByValueStepsSchema.validate([step]);
+          }).toThrowErrorMatchingInlineSnapshot(
+            `"[0]: Cannot provide both \\"lt\\" and \\"lte\\" for the same step."`
+          );
+        });
+
+        it('should invalidate using inverted range - lt', () => {
+          const step: ColorByValueStep = { color, gte: 100, lt: 50 };
+
+          expect(() => {
+            colorByValueStepsSchema.validate([step]);
+          }).toThrowErrorMatchingInlineSnapshot(
+            `"[0]: Inverted range: \\"gte\\" value must be less than the \\"lt\\" value"`
+          );
+        });
+
+        it('should invalidate using inverted range - lte', () => {
+          const step: ColorByValueStep = { color, gte: 100, lte: 50 };
+
+          expect(() => {
+            colorByValueStepsSchema.validate([step]);
+          }).toThrowErrorMatchingInlineSnapshot(
+            `"[0]: Inverted range: \\"gte\\" value must be less than the \\"lte\\" value"`
+          );
+        });
+      });
     });
 
-    it('validates percentage range type', () => {
-      const input: ColorByValueType = {
-        type: 'dynamic',
-        min: 0,
-        max: 100,
-        range: 'percentage',
-        steps: [
-          {
-            type: 'from',
-            from: 0,
-            color: '#ff0000',
-          },
-          {
-            type: 'exact',
-            value: 50,
-            color: '#0000ff',
-          },
-          {
-            type: 'to',
-            to: 75,
-            color: '#00ff00',
-          },
-        ],
-      };
-
-      const validated = allColoringTypeSchema.validate(input);
-      expect(validated).toEqual(input);
-    });
-
-    it('throw on invalid steps sorting order', () => {
-      const input: ColorByValueType = {
-        type: 'dynamic',
-        range: 'absolute',
-        steps: [
-          {
-            type: 'from',
-            from: 0,
-            color: '#ff0000',
-          },
-          {
-            type: 'to',
-            to: 50,
-            color: '#00ff00',
-          },
-          {
-            type: 'exact',
-            value: 75,
-            color: '#0000ff',
-          },
-        ],
-      };
-
-      expect(() => allColoringTypeSchema.validate(input)).toThrow();
-    });
-
-    it('throws on invalid range type', () => {
-      const input = {
-        type: 'dynamic',
-        min: 0,
-        max: 100,
-        range: 'invalid',
-        steps: [],
-      };
-
-      expect(() => allColoringTypeSchema.validate(input)).toThrow();
-    });
+    describe.each<ColorByValueType['range']>(['absolute', 'percentage'])(
+      'range type - %s',
+      (range) => {
+        const baseConfig = freeze<ColorByValueType>({
+          type: 'dynamic',
+          range,
+          steps: [
+            { gte: 0, lt: 50, color: 'red' },
+            { gte: 50, lt: 75, color: 'green' },
+            { gte: 75, lte: 100, color: 'blue' },
```

Comment:
If `lt` of the current step and `gte` of the ext step must be exactly the same, why can't just we drop the `lt` one in this case?

Take this for example, it looks clearer than the one above who leaks some implementation details:

```ts
[
  { gte: 0, color: 'red' },
  { gte: 50, color: 'green' },
  { gte: 75, lte: 100, color: 'blue' }
]
```